File: src/main/java/toker/warbandscripts/panel/repository/ServerConfigurationRepository.java
package toker.warbandscripts.panel.repository;

import toker.warbandscripts.panel.entity.ServerConfiguration;

import java.util.Collection;
import java.util.Optional;

public interface ServerConfigurationRepository extends BaseRepository<ServerConfiguration, Integer> {

    Optional<ServerConfiguration> findByServerIdAndName(Integer serverId, String name);
    Collection<ServerConfiguration> findAllByServerId(Integer serverId);
}

File: src/main/java/toker/panel/repository/ServerConfigurationRepository.kt
package toker.panel.repository

import toker.panel.entity.ServerConfiguration
import java.util.*

interface ServerConfigurationRepository : BaseRepository<ServerConfiguration, Int> {
    fun findByServerIdAndName(serverId: Int, name: String): Optional<ServerConfiguration>
    fun findAllByServerId(serverId: Int): Collection<ServerConfiguration>
}